utils/run_android_ci.py

```diff
@@ -0,0 +1,144 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0.
+
+# Built-in
+import argparse
+import os
+import time
+import datetime
+
+# Following needs to be installed via pip
+import requests # - for uploading files
+import boto3  # - for launching sample
+
+parser = argparse.ArgumentParser(description="Utility script to upload and run Android Device tests on AWS Device Farm for CI")
+parser.add_argument('--run_id', required=True, help="A unique number for each workflow run within a repository")
+parser.add_argument('--run_attempt', required=True, help="A unique number for each attempt of a particular workflow run in a repository")
+parser.add_argument('--project_arn', required=True, help="Arn for the Device Farm Project the apk will be tested on")
+parser.add_argument('--device_pool_arn', required=True, help="Arn for device pool of the Device Farm Project the apk will be tested on")
+
+current_working_directory = os.getcwd()
+build_file_location = current_working_directory + '/sdk/tests/android/testapp/build/outputs/apk/debug/testapp-debug.apk'
+test_file_location = current_working_directory + '/sdk/tests/android/testapp/build/outputs/apk/androidTest/debug/testapp-debug-androidTest.apk'
+
+def main():
+    args = parser.parse_args()
+    run_id = args.run_id
+    run_attempt = args.run_attempt
+    project_arn = args.project_arn
+    device_pool_arn = args.device_pool_arn
+
+    print("Beginning Android Device Farm Setup\n")
+
+    # Create Boto3 client for Device Farm
+    try:
+        client = boto3.client('devicefarm', region_name='us-west-2')
+    except Exception:
+        print("Error - could not make Boto3 client. Credentials likely could not be sourced")
+        return -1
+    print("Boto3 client established")
+
+
+    # Upload the build apk file to Device Farm
+    upload_file_name = 'CI-' + run_id + '-' + run_attempt + '.apk'
+    print('Upload file name: ' + upload_file_name)
+
+    # Setup upload to Device Farm project
+    create_upload_response = client.create_upload(
+        projectArn=project_arn,
+        name=upload_file_name,
+        type='ANDROID_APP'
+    )
+    device_farm_upload_arn = create_upload_response['upload']['arn']
```

Comment:
Does `client.create_upload()` always succeed? For the requests below we check the `['upload']['status']` field

Reply:
The `create_upload` should throw an exception if something is wrong with the request: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/devicefarm/client/create_upload.html

The `upload_status` check using the arn is checking the file's status before proceeding to make sure it's ready for use before the next step. The status in that check mirrors the status received in `create_upload` for that specific arn.